shared/fs/filepreview/index.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import {globalStyles, globalColors, globalMargins, isMobile} from '../../styles'
+import {Box, Button, Icon, Text} from '../../common-adapters'
+import {formatTimeForMessages} from '../../util/timestamp'
+import FolderHeader from '../header/header-container'
+
+type FilePreviewHeaderProps = {
+  title: string,
+  desc: string,
+}
+
+const FilePreviewHeader = ({title, desc}: FilePreviewHeaderProps) => (
+  <Box>
+    <Box style={filePreviewHeaderStyle}>
+      <Text type="BodyBig">{title}</Text>
+      <Text type="BodySmall">{desc}</Text>
+    </Box>
+  </Box>
+)
+
+type FilePreviewProps = {
+  path: Types.Path,
+  meta: Types.PathItemMetadata,
+}
+
+const humanReadableFileSize = meta => {
+  if (!meta) return ''
+  const size = meta.size
+  if (size >= 1024 ** 4) return `${Math.round(size / 1024 ** 4)}tb`
+  if (size >= 1024 ** 3) return `${Math.round(size / 1024 ** 3)}gb`
+  if (size >= 1024 ** 2) return `${Math.round(size / 1024 ** 2)}mb`
+  if (size >= 1024) return `${Math.round(size / 1024)}kb`
+  return '' + size
+}
+
+class FilePreview extends React.PureComponent<FilePreviewProps> {
+  render() {
+    const {path, meta} = this.props
+    const fileName = Types.getPathName(path)
+    let desc = 'loading'
+    if (meta) {
+      desc = 'Modified on ' + formatTimeForMessages(meta.lastModifiedTimestamp)
+      if (meta.lastWriter) desc += ' by ' + meta.lastWriter
+    }
+    // FIXME is using FolderHeader instead of BackButton ok?
```

Comment:
we have HeaderHoc for this already

Reply:
Noted. Thanks for the detailed feedback!

Reply:
We can refactor it later.